### PR TITLE
Admonitions about SSL error on registration

### PR DESCRIPTION
**Pull request by Loquacity**
moved note to tshoot, added xrefs everywhere

https://bugzilla.suse.com/show_bug.cgi?id=1172146

### CODE REVIEW

File: modules/client-configuration/pages/clients-centos.adoc [hunk 1]
```diff
@@ -176,6 +176,14 @@ endif::[]
 For more information, see xref:client-configuration:registration-overview.adoc[].
 
 
+[WARNING]
+====
+Some clients experience SSL errors when you register them in the {productname} {webui} using a bootstrap script.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-centos.adoc:181]: Not some, but CentOS6 also I find "when you register them in the {productname} {webui} using a bootstrap script" confusing. You just "register" systems on {productname} and you can do that from WebUI or using a bootstrap script (or IIRC also from API).

File: modules/client-configuration/pages/clients-centos.adoc [hunk 2]
```diff
@@ -176,6 +176,14 @@ endif::[]
...
+This can make some operations unusable, including package management and bootstrapping.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-centos.adoc:182]: Well, if you can't bootstrap, for sure you can't do this, right? :-D

If would rephrase this to something like:

> CentOS6 clients can not be registered or managed correctly unless {productname} server is configured to support old encryptions. Check ```Support for old clients``` at configuration:tshoot-clients.adoc[].

File: modules/client-configuration/pages/clients-debian.adoc [hunk 3]
```diff
@@ -165,6 +165,14 @@ mgr-create-bootstrap-repo --with-custom-channels
 For {debian}{nbsp}10, select ``debian10-amd64-uyuni`` when prompted.
 
 
+[WARNING]
+====
+Some clients experience SSL errors when you register them in the {productname} {webui} using a bootstrap script.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-debian.adoc:170]: I am not aware of this problem on Debian, not even Debian 9. I think this can go away.

File: modules/client-configuration/pages/clients-opensuse.adoc [hunk 4]
```diff
@@ -120,3 +120,11 @@ mgr-create-bootstrap-repo --with-custom-channels
 ----
 
 For more information on registering your clients, see xref:client-configuration:registration-overview.adoc[].
+
+
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-opensuse.adoc:124]: This is NOT a problem on openSUSE, as any version that had this problem is EoL since years. This can go away.

File: modules/client-configuration/pages/clients-oracle.adoc [hunk 5]
```diff
@@ -151,3 +151,11 @@ endif::[]
 
 {oracle} clients are registered in the same way as all other clients.
 For more information, see xref:client-configuration:registration-overview.adoc[].
+
+
+[WARNING]
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-oracle.adoc:156]: Same as for CentOS. Only Oracle6 has this problem. Rest of my comments for CentOS6 also apply here.

File: modules/client-configuration/pages/clients-rh.adoc [hunk 6]
```diff
@@ -484,6 +484,14 @@ mgr-create-bootstrap-repo --with-custom-channels
 For more information on registering your clients, see xref:client-configuration:registration-overview.adoc[].
 
 
+[WARNING]
+====
+Some clients experience SSL errors when you register them in the {productname} {webui} using a bootstrap script.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-rh.adoc:489]: Same as before.

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 7]
```diff
@@ -219,6 +219,14 @@ You will need to check all the base and child channel log files to be sure that
...
 
-You {es} clients are now ready to be registered.
+Your {es} clients are now ready to be registered.
 
 For more information on registering your clients, see xref:client-configuration:registration-overview.adoc[].
+
+
+[WARNING]
+====
+Some clients experience SSL errors when you register them in the {productname} {webui} using a bootstrap script.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-sleses.adoc:229]: Same as before.

File: modules/client-configuration/pages/clients-ubuntu.adoc [hunk 8]
```diff
@@ -332,3 +334,11 @@ mgr-create-bootstrap-repo --with-custom-channels
 ----
 
 For more information on registering your clients, see xref:client-configuration:registration-overview.adoc[].
+
+
+[WARNING]
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/clients-ubuntu.adoc:339]: Same as Debian. I am not aware about this being a problem for any supported Ubuntu version. This can go away.

File: modules/client-configuration/pages/registration-bootstrap.adoc [hunk 9]
```diff
@@ -69,6 +69,14 @@ For more about custom certificates, see xref:administration:ssl-certs.adoc[].
 ====
 
 
+[WARNING]
+====
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/registration-bootstrap.adoc:73]: We need to change this as well, in the same way we need changes for *6 OS. Maybe we should have the same message here, but with the list of affected OS.

File: modules/client-configuration/pages/registration-bootstrap.adoc [hunk 10]
```diff
@@ -99,7 +107,7 @@ cp bootstrap.sh bootstrap-sles15.sh
 ----
 . Open [path]``bootstrap-sles12.sh`` for modification.
 Scroll down until you can see the text shown below.
-If ``exit 1`` exists in the file, comment it out by typing a hash or pound sign (``#``) at the beginning of the line.
+If ``exit 1`` exists in the file, comment it out by typing a hash or pound sign at the beginning of the line.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/registration-bootstrap.adoc:110]: So I guess we have `pound sign` because: 

> For some people in the US, "pound sign" refers to the symbol # (number sign).

However for non-native (at least for me) pound sign is `£`. So if you do this change, for me it means I can comment with `#` or `£` which is not correct.

File: modules/client-configuration/pages/registration-cli.adoc [hunk 11]
```diff
@@ -113,31 +113,3 @@ systemctl restart salt-minion
 Your newly registered client should now show up within the {webui} under menu:Salt[Keys].
 Accept the [guimenu]``pending`` key to begin management.
 ////
-
-
-
-== SLES{nbsp}11 SSL errors
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/registration-cli.adoc:119]: Shouldn't we have here the same warning you added at the bootstrap script?

File: modules/client-configuration/pages/registration-webui.adoc [hunk 12]
```diff
@@ -40,29 +40,9 @@ To review a package EULA, open the package details page in the {webui}.
...
-----
-SSLProtocol all -SSLv2 -SSLv3
-----
-
-This will need to be done manually on the server, and with a Salt state on the Proxy.
-Restart the [systemitem]``apache`` service on each system after making the changes.
\ No newline at end of file
+[WARNING]
+====
+Some clients experience SSL errors when you register them in the {productname} {webui} using a bootstrap script.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/registration-webui.adoc:45]: Same comments I did for the bootstrap script part.

File: modules/client-configuration/pages/tshoot-clients.adoc [hunk 13]
```diff
@@ -135,3 +135,32 @@ grains:
       - name.one
       - name.two
 ----
+
+
+
+== SSL Errors on Registration
+
+Some clients experience SSL errors when you register them in the {productname} {webui} using a bootstrap script.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/tshoot-clients.adoc:143]: Same comments as before on the other pages. Here we should also have the list of affected OS.

File: modules/client-configuration/pages/tshoot-clients.adoc [hunk 14]
```diff
@@ -135,3 +135,32 @@ grains:
       - name.one
       - name.two
 ----
+
+
+
+== SSL Errors on Registration
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/tshoot-clients.adoc:141]: I'd do this more generic as I told before. So people can go here in the event the Apache config is rolled back to allow only TLS 1.2

File: modules/client-configuration/pages/tshoot-clients.adoc [hunk 15]
```diff
@@ -135,3 +135,32 @@ grains:
...
+
+Some clients experience SSL errors when you register them in the {productname} {webui} using a bootstrap script.
+This can make some operations unusable, including package management and bootstrapping.
+In this case, you will see an error like this:
+
+----
+Repository '<Repository_Name>' is invalid.
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/tshoot-clients.adoc:148]: At least from WebUI this is not the error you see (I think you see this from bootstrap script).

Should we include both? Should we include none?

File: modules/client-configuration/pages/registration-bootstrap.adoc (hunk 10 above)
Comment (Loquacity) [reply modules/client-configuration/pages/registration-bootstrap.adoc:110]: heh, I just took it out because it was messy and it was breaking my code highlighting. I'll add it back (personally I thought "hash or pound" covered it, but I can see the point).

File: modules/client-configuration/pages/tshoot-clients.adoc (hunk 15 above)
Comment (Loquacity) [reply modules/client-configuration/pages/tshoot-clients.adoc:148]: What error do we get in the webui? I think for this section including the error is a good idea.

Comment (juliogonzalez) [reply modules/client-configuration/pages/tshoot-clients.adoc:148]:
```
Rendering SLS 'base:bootstrap' failed: Jinja error: >>> No TLS 1.2 and above for RHEL6 and SLES11. Please check your Apache config. <<< Traceback (most recent call last): File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 392, in render_jinja_tmpl output = template.render(**decoded_context) File "/usr/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render return original_render(self, *args, **kwargs) File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render return self.environment.handle_exception(exc_info, True) File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception reraise(exc_type, exc_value, tb) File "/usr/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise raise value.with_traceback(tb) File "<template>", line 53, in top-level template code File "/usr/lib/python3.6/site-packages/salt/utils/jinja.py", line 211, in jinja_raise raise TemplateError(msg) salt.exceptions.TemplateError: >>> No TLS 1.2 and above for RHEL6 and SLES11. Please check your Apache config. <<< ; line 53 --- [...] {%- if not grains['os_family'] == 'Debian' %} {%- set bootstrap_repo_request = salt['http.query'](bootstrap_repo_url + 'repodata/repomd.xml', status=True, verify_ssl=False) %} {# 901 is a special status code for the TLS issue with RHEL6 and SLE11. #} {%- if bootstrap_repo_request['status'] == 901 %} {{ raise(bootstrap_repo_request['error']) }} <====================== {%- endif %} {%- set bootstrap_repo_exists = (0 < bootstrap_repo_request['status'] < 300) %} bootstrap_repo: file.managed: [...] ---
```
Really ugly, but it's how it is :(

Comment (srbarrios) [reply modules/client-configuration/pages/tshoot-clients.adoc:148]: Also, the message rendered needs to be updated to include the other products, already mentioned in that PR, so CentOS and Oracle.

**Comment by srbarrios** [comment]
> LGTM but @Loquacity, please get approval from @srbarrios as well (in case I am missing something).

I don't see anything else... just what @juliogonzalez mentioned about scp and python as pre-requisite.

**Comment by juliogonzalez** [comment]
> I don't see anything else... just what @juliogonzalez mentioned about scp and python as pre-requisite.

In fact even Perl is required (IIRC only bootstrapping vía WebUI). There are thing we assume will be on an OS when we bootstrap but... it's not always the case (depends on the setup, either done by the user or by the cloud provider).

**Comment by Loquacity** [comment]
> No, wait, it does NOT LGTM.
> 
> The description still talks about the bug for `scp` (https://bugzilla.suse.com/show_bug.cgi?id=1172124) but there are no references here :-)
> 
> And FMPOV we should tackle the bugzilla issue about `scp` on a different PR, it's not even on the title :-)

I read this after I updated this PR, sorry :(

Let me know if I should change it.

File: modules/client-configuration/pages/tshoot-clients.adoc [hunk 16]
```diff
@@ -140,9 +140,53 @@ grains:
...
+----
+
+* Fix: Before registering {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients:
++
+* Ensure you have installed python and scp
```
Comment (juliogonzalez) [inline modules/client-configuration/pages/tshoot-clients.adoc:170]: No, this is wrong. This is not about old clients. It can happen on any client, and IMHO it should be an admonition for all of them, or a requirement, or something similar.

And if you are going to have it as part of this PR, you should edit the PRs title (but FMPOV you should think about this and do it on a separate PR).

Comment (Loquacity) [reply modules/client-configuration/pages/tshoot-clients.adoc:170]: oh, you want it as a completely separate troubleshooting section? I can do that Thurs for you :+1:

Comment (juliogonzalez) [reply modules/client-configuration/pages/tshoot-clients.adoc:170]: I am not sure this belongs to troubleshooting. Shouldn't we have a "Requirements" page that is common to all OS maybe?

Comment (Loquacity) [reply modules/client-configuration/pages/tshoot-clients.adoc:170]: There's a new clients hardware requirements table in the Install Guide (modules/installation/pages/client-requirements.adoc), we could add a software pre-req note as well? Otherwise, we could put it at the beginning of the Client Cfg, in modules/client-configuration/pages/supported-features.adoc ?

Comment (juliogonzalez) [reply modules/client-configuration/pages/tshoot-clients.adoc]: I'd use `modules/installation/pages/client-requirements.adoc`.

Maybe not all customers will read `modules/client-configuration/pages/supported-features.adoc`

Just keep in mind `python` and `scp` are not the package name (those change, are different for different OS). In fact I am not 100% sure about python. I know it's required for salt and salt-ssh, but not so sure about traditional (and I guess python2 or python3 are needed depending on the OS).

Please check with the salt team.

File: modules/client-configuration/pages/clients-centos.adoc [hunk 17]
```diff
@@ -176,6 +176,13 @@ endif::[]
 For more information, see xref:client-configuration:registration-overview.adoc[].
 
 
+[WARNING]
+====
+In order to register and use {centos}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```
Comment (keichwa) [inline modules/client-configuration/pages/clients-centos.adoc:181]:
```suggestion
To register and use {centos}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```
per style guide, iirc.

File: modules/client-configuration/pages/clients-oracle.adoc [hunk 18]
```diff
@@ -93,7 +93,7 @@ oraclelinux8-appstream oraclelinux8-uyuni-client
 The client tools channel provided by [command]``spacewalk-common-channels`` is sourced from {uyuni} and not from {suse}.
 ====
 
- .Procedure: Synchronizing Oracle repositories
+.Procedure: Synchronizing Oracle repositories
```
Comment (keichwa) [inline modules/client-configuration/pages/clients-oracle.adoc:96]:
```suggestion
.Procedure: Synchronizing Oracle Repositories
```

File: modules/client-configuration/pages/clients-oracle.adoc [hunk 19]
```diff
@@ -93,7 +93,7 @@ oraclelinux8-appstream oraclelinux8-uyuni-client
...
  . In the Uyuni Web UI, navigate to Software Manage, and check every Oracle channel.
  . In the Repositories tab, navigate to the Sync subtab, and click Sync Now. You can also create a regular synchronization schedule on this page.
```
Comment (keichwa) [inline modules/client-configuration/pages/clients-oracle.adoc:98]: here and above we want adoc tag soup ;)

File: modules/client-configuration/pages/clients-oracle.adoc [hunk 20]
```diff
@@ -151,3 +151,10 @@ endif::[]
 
 {oracle} clients are registered in the same way as all other clients.
 For more information, see xref:client-configuration:registration-overview.adoc[].
+
+
+[WARNING]
+====
+In order to register and use {oracle}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```
Comment (keichwa) [inline modules/client-configuration/pages/clients-oracle.adoc:158]:
```suggestion
To register and use {oracle}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```

File: modules/client-configuration/pages/clients-rh.adoc [hunk 21]
```diff
@@ -484,6 +484,13 @@ mgr-create-bootstrap-repo --with-custom-channels
 For more information on registering your clients, see xref:client-configuration:registration-overview.adoc[].
 
 
+[WARNING]
+====
+In order to register and use {rhel}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```
Comment (keichwa) [inline modules/client-configuration/pages/clients-rh.adoc:489]:
```suggestion
To register and use {rhel}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```

File: modules/client-configuration/pages/clients-sleses.adoc [hunk 22]
```diff
@@ -219,6 +219,13 @@ You will need to check all the base and child channel log files to be sure that
...
 
-You {es} clients are now ready to be registered.
+Your {es} clients are now ready to be registered.
 
 For more information on registering your clients, see xref:client-configuration:registration-overview.adoc[].
+
+
+[WARNING]
+====
+In order to register and use {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```
Comment (keichwa) [inline modules/client-configuration/pages/clients-sleses.adoc:229]:
```suggestion
To register and use {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```

File: modules/client-configuration/pages/registration-bootstrap.adoc [hunk 23]
```diff
@@ -69,6 +69,13 @@ For more about custom certificates, see xref:administration:ssl-certs.adoc[].
 ====
 
 
+[WARNING]
+====
+In order to register and use {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```
Comment (keichwa) [inline modules/client-configuration/pages/registration-bootstrap.adoc:74]:
```suggestion
To register and use {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```

File: modules/client-configuration/pages/registration-bootstrap.adoc [hunk 24]
```diff
@@ -99,7 +106,7 @@ cp bootstrap.sh bootstrap-sles15.sh
 ----
 . Open [path]``bootstrap-sles12.sh`` for modification.
 Scroll down until you can see the text shown below.
-If ``exit 1`` exists in the file, comment it out by typing a hash or pound sign (``#``) at the beginning of the line.
+If ``exit 1`` exists in the file, comment it out by typing a hash or pound sign (&#35;) at the beginning of the line.
```
Comment (keichwa) [inline modules/client-configuration/pages/registration-bootstrap.adoc:109]: elsewhere you removed it completely, iirc

File: modules/client-configuration/pages/registration-cli.adoc [hunk 25]
```diff
@@ -57,6 +57,15 @@ systemctl restart salt-minion
 salt-key -a '<client>'
 ----
 
+
+[WARNING]
+====
+In order to register and use {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```
Comment (keichwa) [inline modules/client-configuration/pages/registration-cli.adoc:63]:
```suggestion
To register and use {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```

File: modules/client-configuration/pages/registration-webui.adoc [hunk 26]
```diff
@@ -40,29 +40,8 @@ To review a package EULA, open the package details page in the {webui}.
...
-----
-SSLProtocol all -SSLv2 -SSLv3
-----
-
-This will need to be done manually on the server, and with a Salt state on the Proxy.
-Restart the [systemitem]``apache`` service on each system after making the changes.
\ No newline at end of file
+[WARNING]
+====
+In order to register and use {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```
Comment (keichwa) [inline modules/client-configuration/pages/registration-webui.adoc:45]:
```suggestion
To register and use {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```

File: modules/client-configuration/pages/tshoot-clients.adoc [hunk 27]
```diff
@@ -135,3 +135,79 @@ grains:
...
++
+This must be done manually on the server, and with a Salt state on the Proxy, if applicable.
+Restart the [systemitem]``apache`` service on each system after making the changes.
+
+* Result: the client is successfully registered
+////
+
+
+
+In order to register and use {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```
Comment (keichwa) [inline modules/client-configuration/pages/tshoot-clients.adoc:183]:
```suggestion
To register and use {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients, you need to configure the {productname} Server to support older types of SSL encryption.
```

File: modules/client-configuration/pages/tshoot-clients.adoc (hunk 16 above)
Comment (Loquacity) [reply modules/client-configuration/pages/tshoot-clients.adoc]: OK, I'll do that in a new PR, so that we can get this one in.

File: modules/client-configuration/pages/clients-oracle.adoc (hunk 19 above)
Comment (Loquacity) [reply modules/client-configuration/pages/clients-oracle.adoc:98]: Yeah, this will all get changed when I fix these chapters, though, so I'm not too concerned at this point.

File: modules/client-configuration/pages/registration-bootstrap.adoc (hunk 24 above)
Comment (Loquacity) [reply modules/client-configuration/pages/registration-bootstrap.adoc:109]: Yes, but Julio complained :wink:
